src/colvarproxy.h

```diff
@@ -48,11 +48,53 @@ class colvarproxy_system {
   /// Destructor
   virtual ~colvarproxy_system();
 
-  /// \brief Value of the unit for atomic coordinates with respect to
-  /// angstroms (used by some variables for hard-coded default values)
-  virtual cvm::real unit_angstrom() = 0;
+  // **********************************************************************
+  // TODO FIXME unsolved: calls to back-end PBC functions assume back-end length unit!
+  // We can use different unit from back-end in VMD bc using PBC functions from colvarproxy base class
+  // or test and default to internal PBC functions if we have our own units?
+  // Possible performance cost? a couple floating-point ops per coordinate
+  // **********************************************************************
+
+  // Colvars internal units are user specified, because the module exchanges info in unknown
+  // composite dimensions with user input, while it only exchanges quantities of known
+  // dimension with the back-end (length and forces)
+
+  /// \brief Name of the unit system used internally by Colvars (by default, that of the back-end).
+  /// Supported depending on the back-end: real (A, kcal/mol), metal (A, eV), electron (Bohr, Hartree), gromacs (nm, kJ/mol)
+  std::string units;
+
+  /// \brief Request to set the units used internally by Colvars
+  virtual int set_unit_system(std::string const &units, bool colvars_defined) = 0;
+
+  /// \brief Value of 1 Angstrom in the internal (front-end) Colvars unit for atomic coordinates
+  /// * defaults to 0. in the base class; derived proxy classes must set it
+  /// * in VMD proxy, can only be changed when no variables are defined
+  /// as user-defined values in composite units must be compatible with that system
+  cvm::real angstrom_value;
+
+  /// \brief Value of 1 Angstrom in the backend's unit for atomic coordinates
+  virtual cvm::real backend_angstrom_value() = 0;
+
+  /// \brief Value of 1 kcal/mol in the internal Colvars unit for energy
+  cvm::real kcal_mol_value;
+
+  /// \brief Value of 1 kcal/mol in the backend's unit for energy
+  //virtual cvm::real backend_kcal_mol_value() = 0;
```

Comment:
Why are this and the analogous function commented out?

Reply:
Because they are not used now, but would be used in a possible implementation that does perform energy unit conversion - either in any MD engine, or in VMD if we say got forces from IMD. To unclutter the code, I'll remove them now. If needed they'll be easy to add back.